internal/build/buildkit_printer.go

```diff
@@ -0,0 +1,78 @@
+package build
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	digest "github.com/opencontainers/go-digest"
+)
+
+type buildkitPrinter struct {
+	output io.Writer
+}
+
+type vertex struct {
+	digest digest.Digest
+	name   string
+	error  string
+}
+
+type vertexLog struct {
+	vertex digest.Digest
+	msg    []byte
+}
+
+type vertexAndLogs struct {
+	vertex *vertex
+	logs   []*vertexLog
+}
+
+func newBuildkitPrinter(output io.Writer) buildkitPrinter {
+	return buildkitPrinter{
+		output: output,
+	}
+}
+
+func (b *buildkitPrinter) parseAndPrint(vertexes []*vertex, logs []*vertexLog) error {
+	vMap := map[digest.Digest]vertexAndLogs{}
+
+	for _, v := range vertexes {
+		vMap[v.digest] = vertexAndLogs{
+			vertex: v,
+		}
+	}
+
+	for _, l := range logs {
+		if val, ok := vMap[l.vertex]; ok {
+			if len(l.msg) > 0 {
+				vMap[l.vertex] = vertexAndLogs{
+					vertex: val.vertex,
+					logs:   append(val.logs, l),
+				}
+			}
+		}
+	}
+
+	for _, v := range vMap {
+		cmdPrefix := "/bin/sh -c "
```

Comment:
should we leave a note that this won't work in all cases and we will need to be smarter here?